core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.action;
+
+import ch.ethz.ssh2.Connection;
+import ch.ethz.ssh2.Session;
+import ch.ethz.ssh2.StreamGobbler;
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import com.google.common.base.Charsets;
+import com.google.common.base.Strings;
+import com.google.common.io.CharStreams;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.EnumSet;
+import javax.annotation.Nullable;
+
+/**
+ * Action that exports data from Oracle.
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("OracleExport")
+@Description("A Hydrator Action plugin to efficiently export data from Oracle to HDFS or local file system. " +
+  "The plugin uses Oracle's command line tools to export data. The data exported from this tool can then " +
+  "be used in Hydrator pipelines.")
+public class OracleExportAction extends Action {
+  enum SeparatorFormat {
+    CSV, TSV, PSV
+  };
+
+  private final OracleExportActionConfig config;
+
+  public OracleExportAction(OracleExportActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate();
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    String oracleExportCommand;
+    if(Strings.isNullOrEmpty(config.commandToRun)) {
+      oracleExportCommand = buildOracleExportCommand();
+    } else {
+      oracleExportCommand = config.commandToRun;
+    }
```

Comment:
With this, the user must give a command that also creates the necessary SQL. Instead of `config.commandToRun` being used for `oracleExportCommand` it should be used for `sqlPlusSpoolExecutionCommand` within `buildOracleExportCommand`.